docs/source/deployment/netlify.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Deploy a fullstack Apollo app with Netlify
+sidebar_title: Netlify
+description: How to run your API using Netlify Functions on AWS Lambda
+---
+
+[Netlify functions](https://www.netlify.com/docs/functions/), is a new way to deploy server-side code to AWS Lambda alongside your frontend, right from GitHub. In this guide, I'll show you why Netlify's functions is a great way to run your whole Apollo app.
+
+1. Use the same platform to deploy your frontend and API code.
+2. Deploy previews of both your frontend and API on every pull request.
+3. No need to worry about scaling containers due to a serverless architecture.
+
+We'll set up a "Hello World" API and frontend for an Apollo app on Netlify. Let's get started!
+
+## Download and Run the Netlify Starter Kit
+
+Download and run the [official Netlify create-react-app and Lambda starter kit](https://github.com/netlify/create-react-app-lambda). This will set us up with a base for our frontend and API code.
+
+
+```bash
+git clone https://github.com/netlify/create-react-app-lambda.git
+cd create-react-app-lambda
+npm install
+```
+
+Let’s take a quick look at the file structure here:
+
+* **package.json:** our dependencies, shared between client and server code. You can also split them up into separate directories later, depending on your preference.
+* **netlify.toml:** configuration for Netlify and the Netlify functions.
+* **src/:** code for our React app.
+* **src/lambda/:** code that will be deployed to Lambda by Netlify.
+
+In `package.json`, you’ll find scripts to run the frontend and lambda code:
+
+```js
+"scripts": {
+  "start": "react-scripts start",
+  "start:lambda": "netlify-lambda serve src/lambda",
+  "build": "react-scripts build",
+  "build:lambda": "netlify-lambda build src/lambda",
+}
+```
+
+Run the code to see that everything is working. We have to open two terminal windows to run the frontend and functions at the same time:
+
+```bash
+# In the first terminal
+npm run start:lambda
+# In the second terminal
+npm start
+```
+
+If everything looks like it started correctly, let’s add a GraphQL API with Apollo Server!
+
+### Setting up Apollo Server
+
+Netlify functions run on AWS Lambda, so we can use the `apollo-server-lambda` package to easily integrate Apollo Server for our API layer. Let’s install the packages we need for that:
+
+```bash
+npm install --save apollo-server-lambda graphql
+```
+
+Now, we can create a “Hello world” GraphQL API. Let’s put that in a new file in the lambda folder, at `src/lambda/graphql.js`:
+
+```js
+// src/lambda/graphql.js
+const { ApolloServer, gql } = require("apollo-server-lambda");
+
+const typeDefs = gql`
+  type Query {
+    hello: String
+  }
+`;
+
+const resolvers = {
+  Query: {
+    hello: (root, args, context) => {
```

Comment:
@abernix What do you mean by it's not always "root" ? Please can I get a verbose explanation or a link to an article that explains why?

Reply:
It's not `root` because revolvers are called in a cascading fashion.  This is only "`root`" it if it's at the very top of the graph entry point.

What it really points to is the resolver that called it — it's parent — and in the case of the first resolver being called, that parent is the `Query` type.